### PR TITLE
Update input field and dropdown styles to match the design system

### PR DESCRIPTION
**Pull request by robintown**
My original goal was just to solve https://github.com/vector-im/element-web/issues/9582 by updating the label color to whatever our design system says it should be, but I then noticed that [the design specifications in Compound](https://www.figma.com/file/X4XTH9iS2KGJ2wFKDqkyed/Compound?node-id=2039%3A26415) ([screenshot for those without access](https://user-images.githubusercontent.com/558581/169160737-7cc3ab0e-06e0-49ce-b035-7fcf9c352715.png)) don't match our current input fields at all. So, I've overhauled the styles accordingly to bring them out of the Riot era. The label animation has also become considerably smoother now that I've changed it to animate `scale` instead of `font-size`.

This definitely needs thorough design review, since this change spans the entire app.

**Before**

https://user-images.githubusercontent.com/48614497/167968861-6452abd7-3104-4589-8e89-ee5f44757ff2.mp4

https://user-images.githubusercontent.com/48614497/167968134-20d91d72-22ee-4e14-8f45-e77c90601238.mp4

**After**

https://user-images.githubusercontent.com/48614497/167968870-cc8b44ed-52fd-476c-8104-0e99e58a58b5.mp4

https://user-images.githubusercontent.com/48614497/167968136-2c0d1f83-4e7e-4f03-ab01-6ea63793d324.mp4

Closes https://github.com/vector-im/element-web/issues/9582.

---
Here's what your changelog entry will look like:

## ✨ Features
 * Update input field and dropdown styles to match the design system ([\#8566](https://github.com/matrix-org/matrix-react-sdk/pull/8566)). Fixes vector-im/element-web#9582.

### CODE REVIEW

File: res/css/views/elements/_Dropdown.scss [hunk 1]
```diff
@@ -105,24 +108,19 @@ input.mx_Dropdown_option:focus {
...
 }
 
-.mx_Dropdown_menu .mx_Dropdown_option {
-    height: auto;
-    min-height: 35px;
-}
-
 .mx_Dropdown_menu .mx_Dropdown_option_highlight {
-    background-color: $focus-bg-color;
+    background-color: rgba($quaternary-content, 0.2);
```
Comment (t3chguy) [inline res/css/views/elements/_Dropdown.scss:123]: This won't work for custom themes where it'll base itself on the inherited parent theme instead, rgba runs in postcss at build time

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Comment by janogarcia** [comment]
Could we fix the latest deployment? It seems it didn't get deployed to Netlify successfully as the [previous one](https://github.com/matrix-org/matrix-react-sdk/pull/8566#event-6594377904).

**Comment by robintown** [comment]
Closing this since we now presumably want to use the new fields from Compound Web going forward, and it's not worth investing time in making changes to old components

**Pull request closed without merging**